packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -593,6 +593,13 @@ class PipelineController extends CanvasController {
     const pipeline = this.findNodeParentPipeline(nodeID);
     if (pipeline !== undefined) {
       const app_data = prefixedToNested(data);
+      for (const [key, val] of Object.entries(
+        app_data.component_parameters ?? {}
+      )) {
+        if (val === undefined && val === null) {
+          delete app_data.component_parameters[key];
+        }
+      }
```

Comment:
Ah I see what this is doing now. We're actually doing something similar on the Elyra side already https://github.com/elyra-ai/elyra/blob/9ea71730266eee3185631524db4f5e0873c139b3/packages/pipeline-editor/src/PipelineEditorWidget.tsx#L284-L292

But I'm unsure why this is needed with the canvas update? Is the newer version of canvas not dealing with undefined/null values?

Reply:
Not totally sure why this showed up but we had a test for it so I figured I'd just include a fix.

Reply:
super weird but whatever